### PR TITLE
Gutenboarding: Accordion-style plans grid.

### LINKED ISSUE #44201: Plans Grid: change layout to one column
Compared to the current mobile view there will be several changes:
- Features list should be split on 2/3 columns depending on screen width.
- _Popular_ label rounded and centered (not full-width).
- Price moved to the right side.
- Premium plan moved to top and accordion style for the rest.
- A button _Show all plans_ that should expand all the accordions

**Before (mobile)**
Screenshot 2020-07-16 at 11 09 48

**After (desktop)**
Screenshot 2020-07-16 at 11 08 28

### PR DESCRIPTION
**Pull request by yansern**
## Changes proposed in this Pull Request

* Accordion-style plans grid.

**Detailed Changes:**

- We now split the rendering of PlanItem, one for primary plan, one for all other plans (it used to be in a single loop). Primary plan could either mean the default premium plan or the recommended plan based on the features user chosen from the feature picker. See https://github.com/Automattic/wp-calypso/pull/44299/files#diff-ef06ad3b1ea3e3d50b5d84d92780d2adR39-R48.
  - The `badge` param is introduced to customize what to show on the primary plan's badge. When default premium plan is shown, the badge display **Popular**. When a recommended plan is shown, the badge displays **Recommended for you**.
- All plans toggling (Show all plans/collapse all plans button) is now the responsibility of the parent component `PlansTable` instead of child component `PlanItem`, see https://github.com/Automattic/wp-calypso/pull/44299/files#diff-ef06ad3b1ea3e3d50b5d84d92780d2adR50-R63. `onToggleExpandAll` and `allPlansExpanded` param has been removed, see https://github.com/Automattic/wp-calypso/pull/44299/files#diff-ef06ad3b1ea3e3d50b5d84d92780d2adR50-R63.
- "Custom domain not included" is now always shown under Free plan's feature list.
- Added plan descriptions "Best for freelancers, best for online stores, etc.".
- PlanItem component at `plan-tables\plan-item.tsx` is now moved out to `plan-item\index.tsx` with it's own stylesheet specific to the component itself.

## Testing instructions

* Go to `/new`.
* Test plans grid in different viewport sizes, make sure they match similarly to design.
* Test toggling "Show all plans / Collapse all plans" button.
  * Manually expanding/collapsing each of the plan items should cause the toggle button to automatically switch "Show all plans" to "Collapse all plans" (and vice versa).
* Test the domain picker button on the feature list with:
  * No/free domain selected.
  * Paid domain selected.
* Ensure all "Choose" verb has been renamed to "Select".

## Screenshots

Fixes #44201

### CODE REVIEW

File: client/landing/gutenboarding/onboarding-block/plans/index.tsx [hunk 1]
```diff
@@ -73,10 +73,10 @@ const PlansStep: React.FunctionComponent< Props > = ( { isModal } ) => {
 	const header = (
 		<>
 			<div>
-				<Title>{ __( 'Choose a plan' ) }</Title>
+				<Title>{ __( 'Select a plan' ) }</Title>
 				<SubTitle>
 					{ __(
-						'Pick a plan that’s right for you. Switch plans as your needs change. There’s no risk, you can cancel for a full refund within 30 days.'
+						'Pick a plan that’s right for you. There’s no risk, you can cancel for a full refund within 30 days.'
```
Comment (a8ci18n) [inline client/landing/gutenboarding/onboarding-block/plans/index.tsx:79]: :warning: This change will be queued for retranslation. We'll display the English text until then.

**Comment by yansern** [comment]
**Things left to work on:**

1. Full width badge on mobile view.
image

2. Intended audience caption.
image

**Comment by michelleweber** [comment]
Re: copy, the captions should be in sentence case, with a capital letter for the first word only.

Have we ever tested using "recommended" vs. "best" for these?

**Comment by razvanpapadopol** [comment]
Few more things related to design:
1. Increase height of the domain checkmark/button to align with the rest of the items.
Screenshot 2020-07-21 at 16 10 34

2. Apply primary button styles to the buttons at the bottom of _Detailed comparison_ table. They should look similar to the buttons used on plans cards (rounded corners, padding, etc).
Screenshot 2020-07-21 at 15 56 58

**Comment by yansern** [comment]
> Re: copy, the captions should be in sentence case, with a capital letter for the first word only.

Plan description (intended audience caption) is now in sentence case. https://github.com/Automattic/wp-calypso/pull/44299/commits/af57d4d7551261a51d1927acc817d45bc1f1fa5e

> Have we ever tested using "recommended" vs. "best" for these?

On the sidenote, from the figma design I can see there are some additional mechanics around this which is tied to the feature picker.

When a user picks a set of features, the badge for the primary plan will show "Recommended For You".
When a user doesn't pick anything, the badge for the recommended plan will show "Popular".

See: https://github.com/Automattic/wp-calypso/pull/44299/files#diff-ef06ad3b1ea3e3d50b5d84d92780d2adR40-R45

**Comment by yansern** [comment]
> Increase height of the domain checkmark/button to align with the rest of the items.

Fixed with `min-height`. https://github.com/Automattic/wp-calypso/pull/44299/files#diff-3cc6363676acacebd55683fdfe4bfa99R357

> Apply primary button styles to the buttons at the bottom of Detailed comparison table. They should look similar to the buttons used on plans cards (rounded corners, padding, etc).

Fixed. https://github.com/Automattic/wp-calypso/pull/44299/commits/f65b19acf0ff6dcfb1efff7c4bbf1e001825b696

**Comment by yansern** [comment]
I reworked the plans grid again based on closer inspection of the figma design, PR descriptions have been updated & screenshots have been replaced.

File: packages/plans-grid/src/plan-item/index.tsx [hunk 2]
```diff
@@ -91,115 +103,120 @@ function domainMessageStateMachine(
...
 					<div
 						tabIndex={ 0 }
 						role="button"
-						onClick={ () => setIsOpenInternalState( ( open ) => ! open ) }
-						onKeyDown={ ( e ) =>
-							e.keyCode === SPACE_BAR_KEYCODE && setIsOpenInternalState( ( open ) => ! open )
-						}
-						className="plan-item__summary"
+						onClick={ handleToggle }
+						onKeyDown={ ( e ) => e.keyCode === SPACE_BAR_KEYCODE && handleToggle() }
```
Comment (razvanpapadopol) [inline packages/plans-grid/src/plan-item/index.tsx:165]: Why not use also _ENTER_ key to toggle accordion as mentioned in [WAI-ARIA spec](https://www.w3.org/TR/wai-aria-practices/)?

File: packages/plans-grid/src/plans-table/index.tsx [hunk 3]
```diff
@@ -29,32 +31,85 @@ const PlansTable: React.FunctionComponent< Props > = ( {
...
 } ) => {
+	const { __ } = useI18n();
+
 	const supportedPlans = useSelect( ( select ) => select( PLANS_STORE ).getSupportedPlans() );
 	const prices = useSelect( ( select ) => select( PLANS_STORE ).getPrices() );
-	const [ allPlansExpanded, setAllPlansExpanded ] = useState( false );
+
+	// Primary plan
+	// TODO: Feature picker determines recommended plan depending on the features that user choose.
+	const recommendedPlanSlug = null;
```
Comment (razvanpapadopol) [inline packages/plans-grid/src/plans-table/index.tsx:41]: This should be passed as a prop to plans-grid package and then passed to this component.

File: packages/plans-grid/src/plans-table/index.tsx [hunk 4]
```diff
@@ -29,32 +31,85 @@ const PlansTable: React.FunctionComponent< Props > = ( {
...
+	const popularPlanSlug = 'value_bundle';
```
Comment (razvanpapadopol) [inline packages/plans-grid/src/plans-table/index.tsx:42]: We should get this value from the plans data-store.
`useSelect( ( select ) => select( PLANS_STORE ).getDefaultPaidPlan() );`

File: packages/plans-grid/src/plans-table/index.tsx (hunk 3 above)
Comment (yansern) [reply packages/plans-grid/src/plans-table/index.tsx:41]: Coming soon!

File: packages/plans-grid/src/plan-item/index.tsx (hunk 2 above)
Comment (yansern) [reply packages/plans-grid/src/plan-item/index.tsx:165]: And also with focus outline (not sure if there is at the moment). Another PR to fix A11y issues.

File: packages/plans-grid/src/plans-table/index.tsx (hunk 4 above)
Comment (yansern) [reply packages/plans-grid/src/plans-table/index.tsx:42]: Going to roll this together with another PR coming up where `useRecommendedPlan()` is used.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4128953

Thank you @yansern for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4128953).

**Pull request merged**